ValveResourceFormat/IO/HammerMeshBuilder.cs

```diff
@@ -72,12 +76,51 @@ public void ComputeHalfEdgeStructure()
                 //generate inner half edge loop
                 for (int j = 0; j < face.indices.Count; j++)
                 {
-                    var v1 = face.indices[j];
-                    var v2 = face.indices[(j + 1) % face.indices.Count];
+                    var v1idx = face.indices[j];
+                    var v2idx = face.indices[(j + 1) % face.indices.Count];
+
+                    var v1 = Vertices[v1idx];
+                    var v2 = Vertices[v2idx];
 
-                    var he = new HalfEdge { origVert = v1, destVert = v2, face = i };
                     var heIndex = initHalfEdgeCount + j;
 
+                    //if the current internal half edge were trying to add already exists something is wrong
+                    //break out of the loop (discard face)
+                    try
+                    {
+                        VertsToEdgeDict.Add(new Tuple<int, int>(v1idx, v2idx), heIndex);
```

Comment:
if (!VertsToEdgeDict.TryAdd instead of try catch

Reply:
I'll trust you on this one but out of curiosity what's the diff?

Reply:
That's the correct way of doing it, without relying on Add to throw on duplicate key.

Reply:
That seems like it's just documenting the behavior.